core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/ClusterBootstrapper.java

```diff
@@ -349,6 +363,37 @@ public void bootstrapNewNodes(Long stackId, Set<String> upscaleCandidateAddresse
         }
     }
 
+    public void rotateSaltPassword(Stack stack) {
+        if (!stack.isAvailable()) {
+            throw new BadRequestException("Rotating salt password is only available for stacks in available status");
+        }
+        if (!isChangeSaltuserPasswordSupported(stack)) {
+            throw new BadRequestException("Rotating salt password is not supported with your image version, " +
+                    "please upgrade to an image with salt-bootstrap version >= 0.13.6 (you can find this information in the image catalog)");
+        }
+        try {
+            SecurityConfig securityConfig = securityConfigService.findOneByStackId(stack.getId())
```

Comment:
I would prefer having `getOneByStackId` in `SecurityConfigService` so it could be reused